### PR TITLE
Dashboard: Fix workaround for stock card

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #13641 

## Description

For some stores, fetched variation reports return parent ID 0. When working with the stock card, I added a workaround to copy the product ID from stock reports to variations parent IDs. The workaround used the incorrect field, so it didn't work properly.

This PR fixes the workaround by copying the `parentID` field instead of `productID` field. This ensures that variation reports have the correct product IDs, which is necessary to navigate from the stock card to the variation detail screen.

## Steps to reproduce

- Log in to a store with parent ID being 0 for variation reports. I'm unsure which plugin causes this issue - maybe @rachelmcr  can give you access to her store for to test.
- Enable the Stock card on the dashboard.
- Select any variation item on the card.
- Confirm that the variation details are loaded correctly.

## Screenshots

https://github.com/user-attachments/assets/88bd80cf-b812-4c9c-aa8d-083e5229753a

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/218">20.0 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR contains changes to <code>RELEASE-NOTES.txt</code>.<br />
Note that these changes won't affect the final version of the release notes as this version is in code freeze.<br />
Please, get in touch with a release manager if you want to update the final release notes.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by selanthiraiyan — APPROVED** [review]
#### Test store setup
I tested by mocking `wc-analytics/reports/variations` endpoint response using proxyman to always return `product_id` as `0`.

---

#### Testing results 
I tapped on both variations and parent products listed in the stock card and their detail pages were loaded correctly. ✅ 

The product details pages load correctly on both iPhone and iPad. ✅ 


https://github.com/user-attachments/assets/f33ba597-ae4d-42f3-99e2-e63c4b5d6237



---
:shipit:

**Comment by selanthiraiyan** [comment]
@itsmeichigo 👋 Consider adding internal release notes to this PR and [the Inbox PR](https://github.com/woocommerce/woocommerce-ios/pull/13645) for testing purposes.

**Comment by itsmeichigo** [comment]
@selanthiraiyan Release notes updated in 78b16241357902d663b2b0e6cdabc5d2fb0db78f. I'm enabling auto-merge now.
@rachelmcr Could you prepare a new beta build after this PR is merged? Thanks!

**Comment by rachelmcr** [comment]
Thanks for the heads up! New beta build 20.0.0.1 is released with this change 👍

**Pull request merged**
